prow/hook/deployment.yaml

```diff
@@ -8,19 +8,43 @@ metadata:
   annotations:
     wave.pusher.com/update-on-config-change: "true"
 spec:
+  minReadySeconds: 10
   strategy:
     type: RollingUpdate
     rollingUpdate:
       maxSurge: 2
       maxUnavailable: 0
-  minReadySeconds: 10
+  selector:
+    matchLabels:
+      app: hook
   template:
     metadata:
       labels:
         app: hook
     spec:
       serviceAccountName: "hook"
       terminationGracePeriodSeconds: 180
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - deck
```

Comment:
Should this perhaps not be `hook` rather than `deck`? (And same for the one below).